apollo-router/src/apollo_telemetry.rs

```diff
@@ -285,13 +285,19 @@ impl SpanExporter for Exporter {
          * Process the batch
          */
         for span in batch.iter().filter(|span| span.name == "graphql_request") {
-            tracing::debug!(%span.name, ?span.start_time, ?span.end_time);
-            tracing::debug!("span: {:?}", span);
+            // Time may wander and if we ever receive a span where start_time >
+            // endtime, then we should just drop the span and continue processing
+            // the rest of the batch
+            if span.start_time > span.end_time {
```

Comment:
I think tracing guarantees that end is always higher than start (but I can't remember where I saw it)

Reply:
> I think tracing guarantees that end is always higher than start (but I can't remember where I saw it)

I feel like it should, but SystemTime definitely doesn't and I can't find a reference either. So, better safe than sorry...

Reply:
I've re-worked this a little to make it clear that we expect to be able to calculate a duration from the start and the end and if we can't we are going to ignore the span.